king_phisher/client/windows/campaign_import.py

```diff
@@ -0,0 +1,439 @@
+				if new_id is None:
+					nods.attrib['type'] = 'null'
+
+	def _get_keys_values(self, element):
+		keys = []
+		rows = []
+		for subelements in element:
+			values = []
+			for node in subelements:
+				keys.append(node.tag) if node.tag not in keys else None
+				if node.attrib['type'] in ('datetime', 'date', 'time'):
+					values.append(serializers.from_elementtree_element(node).replace(tzinfo=None))
+				else:
+					values.append(serializers.from_elementtree_element(node))
+			rows.append(tuple(values))
+		return keys, rows
+
+	def interact(self):
+		self.window.set_position(Gtk.WindowPosition.CENTER_ON_PARENT)
+		self.window.show_all()
+
+	def signal_entry_change(self, _):
+		"""
+		When there is a change in the campaign entry field it will check to see if the name is already in use.
+		If it is not in use it will change the Sensitivity of the Import Campaign Button to allow the user to
+		start the import process.
+		"""
+		if not self.campaign_info:
+			return
+		if not self._check_campaign_name(self.entry_campaign_name.get_text()):
+			self.button_import_campaign.set_sensitive(False)
+			return
+		self.button_import_campaign.set_sensitive(True)
+		return
+
+	def signal_multi_open_xml_file(self, _):
+		if not self.select_button.get_property('sensitive'):
+			return
+		self.select_xml_campaign()
+
+	def signal_multi_close_import_campaign(self, _, __):
+		"""
+		Checks to make sure the import campaign thread is closed before closing.
+		"""
+		if not self.campaign_info:
+			return False
+		if not self.thread_import_campaign:
+			return False
+		if not self.thread_import_campaign.is_alive():
+			return False
+		response = gui_utilities.show_dialog(
+			Gtk.MessageType.QUESTION,
+			'Do you want to cancel importing the campaign?',
+			self.window,
+			message_buttons=Gtk.ButtonsType.YES_NO
+		)
+		print(response)
+		if not response or response == -9:
+			return True
+
+		self.thread_import_campaign.stop()
+		self.thread_import_campaign.join()
+		self.remove_import_campaign()
+
+	def failed_import_action(self):
+		response = gui_utilities.show_dialog(
+			Gtk.MessageType.QUESTION,
+			'Import Failed, do you want to delete this campaign?',
+			self.window,
+			message_buttons=Gtk.ButtonsType.YES_NO
+		)
+		return response
+
+	def remove_import_campaign(self):
+		"""
+		Used to delete the imported campaign on failure or early exit of the import window, if the user
+		selects to have it removed.
+		"""
+		campaign_id = self.campaign_info.find('id').text
+		campaign_name = self.campaign_info.find('name').text
+		campaign_check = self.rpc('db/table/get', 'campaigns', campaign_id)
+		if not campaign_check:
+			return
+		if campaign_name == campaign_check['name']:
+			self.rpc('db/table/delete', 'campaigns', campaign_id)
+			self.logger.info('Deleted Campaign {}'.format(campaign_id))
+
+	def signal_import_button(self, _):
+		"""
+		Function called when the user starts the import campaign process, It will create a new thread,
```

Comment:
`...process, It will create a new thread...` is improper grammar, looks like a period is missing.